### ISSUE TITLE
Building on Linux 

### ISSUE THREAD

**Issue by alxvth** [biovault/HDILib#27]
The cmake that sets up the project uses the variable `HDI_EXTERNAL_FLANN_INCLUDE_DIR` for referencing the FLANN include directory. This variable is sets in the automatic CD/CI build but not when following the install instructions given in the README.

The user has to set this path manually:
`cmake -DCMAKE_BUILD_TYPE=Release -DHDI_EXTERNAL_FLANN_INCLUDE_DIR=<PATH/TO/FLANN/> ..`
with <PATH/TO/FLANN> might be /usr/include/ 

Also, since CDRoaring dependencies were recently removed, the install-dependencies.sh does not need to build it anymore, right?

**Comment by alxvth** [comment]
Update for building with Ubuntu 22.10 and gcc 12, and the current cmake

```
sudo apt install libflann-dev
mkdir build
cd build
cmake -DCMAKE_BUILD_TYPE=Release -Dflann_INCLUDE_DIR=/usr/include/ ..
make && make install
```

**Comment by alxvth** [comment]
Addressed with #33:

```bash
sudo apt-get -y install libflann-dev liblz4-dev
cmake  -S . -B build -DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_PREFIX=install -DENABLE_PID=ON -G "Unix Makefiles"
cmake --build build --config Release --target install
```
